cmd/habitat/community/executor.go

```diff
@@ -1,54 +1,152 @@
 package community
 
 import (
-	"encoding/base64"
 	"encoding/json"
-	"fmt"
-	"path/filepath"
 
 	"github.com/eagraf/habitat/cmd/habitat/community/state"
+	"github.com/eagraf/habitat/cmd/habitat/node"
 	"github.com/eagraf/habitat/cmd/habitat/procs"
-	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/structs/community"
 	"github.com/rs/zerolog/log"
 )
 
+type TransitionExecutor func(update *state.StateUpdate) error
+
 type CommunityExecutor struct {
-	processManager *procs.Manager
+	node *node.Node
 }
 
-func NewCommunityExecutor(p *procs.Manager) *CommunityExecutor {
+func NewCommunityExecutor(n *node.Node) *CommunityExecutor {
 	return &CommunityExecutor{
-		processManager: p,
+		node: n,
+	}
+}
+
+// GetTransitionExecutor maps transition types to executor functions
+func (e *CommunityExecutor) GetTransitionExecutor(transitionType string) TransitionExecutor {
+	switch transitionType {
+	case state.TransitionTypeInitializeIPFSSwarm:
+		return e.InitializeIPFSSwarm
+	case state.TransitionTypeStartProcess:
+		return e.StartProcess
+	case state.TransitionTypeStopProcess:
+		return e.StopProcess
+	case state.TransitionTypeStartProcessInstance:
+		return e.StartProcessInstance
+	case state.TransitionTypeStopProcessInstance:
+		return e.StartProcessInstance
+	default:
+		return nil
 	}
 }
 
 func (e *CommunityExecutor) Execute(update *state.StateUpdate) {
 	log.Info().Msgf("executing %s state transition", update.TransitionType)
-	newState, err := update.State()
+
+	transitionExecutor := e.GetTransitionExecutor(update.TransitionType)
+	if transitionExecutor != nil {
+		err := transitionExecutor(update)
+		if err != nil {
+			log.Error().Err(err).Msgf("error executing %s", update.TransitionType)
+		}
+	}
+}
+
+func (e *CommunityExecutor) InitializeIPFSSwarm(update *state.StateUpdate) error {
+	/*newState, err := update.State()
```

Comment:
why is this commented out now?

Reply:
woops forgot to undo this

Reply:
ended up changing how ipfs-driver is started a bit